crates/vault-registry/src/lib.rs

```diff
@@ -861,22 +929,36 @@ impl<T: Config> Module<T> {
     }
 
     /// Automatically liquidates all vaults under the secure threshold
-    fn liquidate_undercollateralized_vaults() -> u32 {
+    pub fn liquidate_undercollateralized_vaults(
+        liquidation_target: LiquidationTarget,
+    ) -> (u32, Vec<(T::AccountId, DOT<T>)>) {
+        // TODO: report system undercollateralization to security
         let mut num_vaults = 0u32;
         let liquidation_threshold = <LiquidationCollateralThreshold<T>>::get();
-        // TODO: report system undercollateralization to security
+        let mut amounts_slashed = Vec::new();
+
         for (vault_id, vault) in <Vaults<T>>::iter() {
             num_vaults = num_vaults.saturating_add(1);
-            if Self::is_vault_below_liquidation_threshold(&vault, liquidation_threshold).unwrap_or(false) {
-                let _ = Self::liquidate_vault(&vault_id);
+            if Self::is_liquidation_target(&vault_id, &liquidation_target) {
+                if Self::is_vault_below_liquidation_threshold(&vault, liquidation_threshold).unwrap_or(false) {
+                    if let Some(to_slash) = Self::liquidate_vault(&vault_id).ok() {
+                        amounts_slashed.push((vault_id.clone(), to_slash));
+                    }
+                }
             }
         }
-        num_vaults
+        (num_vaults, amounts_slashed)
+    }
+
+    fn is_liquidation_target(vault_id: &T::AccountId, liquidation_target: &LiquidationTarget) -> bool {
```

Comment:
Motivation for this makes sense but might be cleaner with separate methods, just to note for @sander2 to review for upcoming OCW optimizations.